### PR TITLE
Call Flashinfer `mm_fp8` for per-tensor FP8 GEMMs on SM100

### PR DESCRIPTION
**Pull request by b8zhong**
## Motivation

It would be more desirable to update the CUTLASS kernel (note), but Flashinfer has autotune between cuBLAS. Call the bmm API with B = 1, only because it's not properly wired to mm_fp8. It also supports autotune between cuBLAS and cuDNN, in addition to it's CUTLASS path.

Note: currently, the CUTLASS path is not a real per-tensor GEMM, it will a weird duplicated scale application on the outputs in the epilogue (it was original designed for channel-wise).

## Modifications

Replace it on SM100.

## Accuracy Tests

```
CUDA_VISIBLE_DEVICES=4,5,6,7 python3 -m sglang.launch_server \
    --model-path nvidia/NVIDIA-Nemotron-3-Ultra-550B-A55B-NVFP4 \
    --trust-remote-code \
    --tp 4 \
    --ep 4 \
    --attention-backend trtllm_mha \
    --mamba-scheduler-strategy extra_buffer \
    --max-running-requests 512 \
    --reasoning-parser nemotron_3 \
    --tool-call-parser qwen3_coder
```

```
python3 benchmark/gsm8k/bench_sglang.py --num-shots 8 --num-questions 1209 --parallel 128 --platinum
Loading GSM8K Platinum dataset from HuggingFace...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1209/1209 [01:12<00:00, 16.59it/s]
Accuracy: 0.984
Invalid: 0.000
Latency: 72.872 s
Output throughput: 1819.568 token/s
```

## Speed Tests and Profiling

Collected with `flashinfer.testing.bench_gpu_time_with_cupti`
image

For example, in the Flashinfer CUTLASS epilogue is now Sm90ScalarBroadcast, instead of Sm90RowBroadcast in sgl-kernel

And now, it will also call nvjet GEMMs (qq means FP8 x FP8)
Screenshot 2026-06-15 at 12 07 21 PM

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #27571725289](https://github.com/sgl-project/sglang/actions/runs/27571725289)
Latest PR Test (Extra): :x: [Run #27571725011](https://github.com/sgl-project/sglang/actions/runs/27571725011)

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 1]
```diff
@@ -571,8 +575,7 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
             layer.weight, layer.weight_scale, layer.logical_widths
         )
         layer.weight = Parameter(quantized_weight.t(), requires_grad=False)
-        # cutlass sgl-kernel only supports per-channel scale
-        if self.cutlass_fp8_supported:
```
Comment (Fridge003) [inline python/sglang/srt/layers/quantization/modelopt_quant.py:575]: Can we open a future PR that cleans all the cutlass_fp8_supported flag. 
We can replace its usage with a new --fp8-gemm-backend

For this flashinfer gemm kernel, we might also create a new fp8 gemm backend for it

**Review by Fridge003 — APPROVED** [review]
